Katello/katello
# Fixes #29125 - Only migrate use_pulp_2_for_content_type types

This PR ensures that `katello:pulp3_migration` and `katello:pulp3_content_switchover` only migrate repository types that have `:use_pulp_2_for_content_type:` as `true`.

To test:
1) Make a Pulp 3 development box
2) Apply this PR
3) Add some repo types (file and docker perhaps) under `:use_pulp_2_for_content_type:` in `foreman/config/settings.plugins.d/katello.yaml` and make their values `true`
4) Stick a `binding.pry` after line 8 of `lib/katello/tasks/pulp3_content_switchover.rake`
5) Run `bundle exec rake katello:pulp3_content_switchover`
6) Check that `content_types` matches what you put in `katello.yaml`
7) Stick a `binding.pry` on the first line of `def create_migrations`
8) Ensure that `@repository_types` returns the types under `:use_pulp_2_for_content_type:`
